clients/python-client/python_client/kuberay_cluster_api.py

```diff
@@ -139,43 +146,68 @@ def get_ray_cluster_status(self, name: str, k8s_namespace: str = "default", time
                     log.error("error fetching custom resource: {}".format(e))
                     return None
 
-            if resource["status"]:
-                    return resource["status"]
+            if resource and "status" in resource and resource["status"]:
+                return resource["status"]
             else:
                 log.info("raycluster {} status not set yet, waiting...".format(name))
                 time.sleep(delay_between_attempts)
                 timeout -= delay_between_attempts
 
-        log.info("raycluster {} status not set yet, timing out...".format(name))
+        log.info("timed out waiting for raycluster {} status".format(name))
         return None
 
-    def wait_until_ray_cluster_running(self, name: str, k8s_namespace: str = "default", timeout: int=60, delay_between_attempts: int = 5) -> bool:
-        """Get a specific Ray cluster in a given namespace.
+    def wait_until_ray_cluster_running(
+        self,
+        name: str,
+        k8s_namespace: str = "default",
+        timeout: int = 60,
+        delay_between_attempts: int = 5,
+    ) -> bool:
+        """Wait until a Ray cluster is in ready state.
+
+        This method waits for the cluster to have a state field with value 'ready'.
 
         Parameters:
-        - name (str): The name of the Ray cluster custom resource. Defaults to "".
+        - name (str): The name of the Ray cluster custom resource.
         - k8s_namespace (str, optional): The namespace in which to retrieve the Ray cluster. Defaults to "default".
-        - timeout (int, optional): The duration in seconds after which we stop trying to get status. Defaults to 60 seconds.
-        - delay_between_attempts (int, optional): The duration in seconds to wait between attempts to get status if not set. Defaults to 5 seconds.
-
-
+        - timeout (int, optional): The duration in seconds after which we stop trying. Defaults to 60 seconds.
+        - delay_between_attempts (int, optional): The duration in seconds to wait between attempts. Defaults to 5 seconds.
 
         Returns:
-            Bool: True if the raycluster status is Running, False otherwise.
-
+            bool: True if the raycluster status is 'ready', False otherwise.
         """
-        status = self.get_ray_cluster_status(name, k8s_namespace, timeout, delay_between_attempts)
-
-        #TODO: once we add State to Status, we should check for that as well  <if status and status["state"] == "Running":>
-        if status and status["head"] and status["head"]["serviceIP"]:
-            return True
-
-        log.info("raycluster {} status is not running yet, current status is {}".format(name, status["state"] if status else "unknown"))
-        return False
-
+        while timeout > 0:
+            status = self.get_ray_cluster_status(
+                name, k8s_namespace, timeout, delay_between_attempts
+            )
 
```

Comment:
Mostly formatting in this file, apart from this change. The the status field for RayCluster has changed since this was implemented, and so the old version of this function did not actually wait for the Cluster to be ready.